homeassistant/components/sensor/skybeacon.py

```diff
@@ -0,0 +1,181 @@
+
+example:
+sensor:
+  - platform: skybeacon
+    mac: 'F7:BE:12:02:47:31'
+    name: 'living room'
+"""
+
+REQUIREMENTS = ['pygatt>=3.0.0']
+
+import logging
+import pygatt
+import time
+import threading
+
+from datetime import timedelta
+from uuid import UUID
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_MAC, TEMP_CELSIUS, STATE_UNKNOWN)
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from pygatt.backends import Characteristic
+from pygatt.exceptions import BLEError, NotConnectedError, NotificationTimeout
+
+CONNECT_LOCK = threading.Lock()
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DEVICE = 'device'
+ATTR_MODEL = 'model'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MAC, default=""): cv.string,
+    vol.Optional(CONF_NAME, default=""): cv.string,
+})
+
+
+BLE_TEMP_UUID = '0000ff92-0000-1000-8000-00805f9b34fb'
+BLE_TEMP_HANDLE = 0x24
+SKIP_HANDLE_LOOKUP = True
+
+CONNECT_TIMEOUT = 30
+
+CACHED_CHR = Characteristic(BLE_TEMP_UUID, BLE_TEMP_HANDLE)
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor."""
+    name = config.get(CONF_NAME)
+    mac = config.get(CONF_MAC)
+    _LOGGER.error("setting up..")
+    mon = Monitor(hass, mac, name)
+    add_devices([SkybeaconTemp(name, mon)])
+    add_devices([SkybeaconHumid(name, mon)])
+    mon.start()
+
+
+class SkybeaconHumid(Entity):
+    """Representation of a humidity sensor."""
+
+    def __init__(self, name, mon):
+        """Initialize a sensor."""
+        self.mon = mon
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.mon.humid
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return "%"
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_DEVICE: "SKYBEACON",
+            ATTR_MODEL: 1,
+        }
+
+class SkybeaconTemp(Entity):
+    """Representation of a temperature sensor."""
+
+    def __init__(self, name, mon):
+        """Initialize a sensor."""
+        self.mon = mon
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.mon.temp
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return TEMP_CELSIUS
+
+    @property
+    def state_attributes(self):
```

Comment:
rename this to `device_state_attributes`. `state_attributes` are reserved for component abstract base classes.